### PR TITLE
Allow variable binding also at params file

### PR DESCRIPTION
**Pull request by jordeu**
Adds the option to alse use variables (following the format `${variable_name}`) when defining parameter values at the JSON or YAML parameters file.

### CODE REVIEW

File: modules/nextflow/src/test/groovy/nextflow/config/ConfigBuilderTest.groovy [hunk 1]
```diff
@@ -475,12 +479,13 @@ class ConfigBuilderTest extends Specification {
...
-        def result = new ConfigBuilder().setOptions(opt).setCmdRun(run).buildGivenFiles(file)
+        def baseDir = Files.createTempDirectory('assets')
+        def result = new ConfigBuilder().setOptions(opt).setCmdRun(run).setBaseDir(baseDir).buildGivenFiles(file)
 
         then:
         result.params.alpha == 'Hola'   // <-- this comes from the CLI
         result.params.beta == 'Mundo'   // <-- this comes from the CLI as well
-        result.params.omega == 'Last'   // <-- this comes from the params-file
+        result.params.omega == "${baseDir.toUriString()}/data/omega.txt"   // <-- this comes from the params-file and replaces parameters
         result.params.gamma == "I'm gamma"   // <-- from the config
```
Comment (pditommaso) [inline modules/nextflow/src/test/groovy/nextflow/config/ConfigBuilderTest.groovy:489]: Here the `baseDir` should not be resolved to the actual value?

Comment (jordeu) [reply modules/nextflow/src/test/groovy/nextflow/config/ConfigBuilderTest.groovy:489]: Yes, it is resolved to the actual value. Notice that I'm using double quotes. At this test `baseDir` is a temporary directory that I cannot hardcode beforehand, this is why I'm using the variable itself.

**Comment by abhi18av** [comment]
I guess, this request https://github.com/nextflow-io/nextflow/issues/1860 can now be closed?

**Comment by jordeu** [comment]
> I guess, this request #1860 can now be closed?
Yes, the only thing is that this implementation do not support variables without curly brackets. Only formatted like this `${baseDir}`.

**Comment by abhi18av** [comment]
That should be fine I think since this is explicit. We could include this info in a `Nextflow config (or params) best practices` section.

**Pull request merged**